standalone-dns-proxy/cmd/standalonednsproxy.go

```diff
@@ -0,0 +1,578 @@
+
+	ipToEndpointIdCache map[string]uint64
+}
+
+// NewStandaloneDNSProxy creates a new standalone DNS proxy
+func NewStandaloneDNSProxy(args *StandaloneDNSProxyArgs) (*StandaloneDNSProxy, error) {
+	if args.toFqdnServerPort == 0 {
+		log.Error("toFqdnServerPort is 0")
+		return nil, errors.New("toFqdnServerPort is 0")
+	}
+
+	return &StandaloneDNSProxy{
+		args:                args,
+		ipToIdentityCache:   make(map[string]uint32),
+		ipToEndpointIdCache: make(map[string]uint64),
+	}, nil
+}
+
+func (sdp *StandaloneDNSProxy) StopStandaloneDNSProxy() error {
+	if sdp.DNSProxy != nil {
+		sdp.DNSProxy.Cleanup()
+	}
+
+	err := sdp.closeConnection()
+	if err != nil {
+		log.WithError(err).Error("Failed to close connection")
+		return err
+	}
+	return nil
+}
+
+// CreateClient creates a client for the cilium agent connection
+// 1. It checks if connection is created, if not it returns an error and triggers the cilium agent connection trigger
+// 2. Else it creates the client
+// 3. If the policy state stream is not created, it creates the stream
+// Note: This function is called with a mutex lock in the caller function because there can be multiple triggers trying to
+// create the stream at the same time
+func (sdp *StandaloneDNSProxy) CreateClient(ctx context.Context) error {
+	var err error
+	defer func() {
+		if err != nil {
+			log.WithError(err).Error("Failed to start cilium agent connection")
+			sdp.closeConnection()
+			sdp.ciliumAgentConnectionTrigger.TriggerWithReason("Failed to start cilium agent connection")
+		}
+	}()
+
+	if sdp.connection == nil {
+		log.Error("Connection is nil")
+		return fmt.Errorf("connection is nil")
+	}
+
+	// Create the client
+	sdp.Client = standalonednsproxy.NewFQDNDataClient(sdp.connection)
+
+	if sdp.policyStateStream == nil {
+		err = sdp.createStreamPolicyStateStream(ctx)
+		if err != nil {
+			log.WithError(err).Error("Failed to create subscription stream")
+			return err
+		}
+	}
+	log.Debugf("Successfully created client for Cilium agent")
+
+	return nil
+}
+
+// ConnectToCiliumAgent creates a connection to the cilium agent
+// It returns an error if the connection is not successful and triggers the cilium agent connection trigger
+func (sdp *StandaloneDNSProxy) ConnectToCiliumAgent() error {
+	var err error
+	defer func() {
+		if err != nil {
+			log.Errorf("Failed to connect to cilium agent: %v", err)
+			sdp.ciliumAgentConnectionTrigger.TriggerWithReason("Failed to connect to cilium agent")
+		}
+	}()
+
+	if sdp.connection != nil {
+		return nil
+	}
+
+	var opts []grpc.DialOption
+	opts = append(opts, grpc.WithInsecure())
+	opts = append(opts, grpc.WithBlock())
+	opts = append(opts, grpc.WithKeepaliveParams(kacp))
+
+	address := fmt.Sprintf("localhost:%d", sdp.args.toFqdnServerPort)
+
+	log.Infof("Connecting to server %v", address)
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*5) // 5 seconds timeout
+	defer cancel()
+
+	conn, err := grpc.DialContext(ctx, address, opts...)
```

Comment:
IIRC `grpc.Dial` and `grpc.DialContext` are [deprecated](https://github.com/grpc/grpc-go/blob/v1.71.0/clientconn.go#L239) - for net new code, I'd not expect usage of deprecated functions.

Reply:
Updated to use `NewClient`